From: Luc Verhaegen <libv@skynet.be>
Date: Mon, 13 Oct 2008 20:28:34 +0000 (+0200)
Subject: Xv: Fix build for big endian.
X-Git-Url: http://gitweb.freedesktop.org/?p=xorg/driver/xf86-video-radeonhd;a=commitdiff;h=e7abdfc07e16c961767dbff72ddbb12cb5af9677

Xv: Fix build for big endian.

Trivial replacement of pScrn -> rhdPtr in R5xxXvCopyPlanar when setting
SURFACE control for big endian.
---

--- a/src/rhd_video.c
+++ b/src/rhd_video.c
@@ -497,8 +497,8 @@ R5xxXvCopyPlanar(RHDPtr rhdPtr, CARD8 *s
 		 CARD16 dstPitch, CARD16 h, CARD16 w)
 {
 #if X_BYTE_ORDER == X_BIG_ENDIAN
-    CARD32 val = RHDRegRead(pScrn, R5XX_SURFACE_CNTL);
-    RHDRegWrite(pScrn, R5XX_SURFACE_CNTL,
+    CARD32 val = RHDRegRead(rhdPtr, R5XX_SURFACE_CNTL);
+    RHDRegWrite(rhdPtr, R5XX_SURFACE_CNTL,
 		(val | R5XX_NONSURF_AP0_SWP_32BPP) & ~R5XX_NONSURF_AP0_SWP_16BPP);
 #endif
 
@@ -507,7 +507,7 @@ R5xxXvCopyPlanar(RHDPtr rhdPtr, CARD8 *s
 
 #if X_BYTE_ORDER == X_BIG_ENDIAN
     /* restore byte swapping */
-    RHDRegWrite(pScrn, R5XX_SURFACE_CNTL, val);
+    RHDRegWrite(rhdPtr, R5XX_SURFACE_CNTL, val);
 #endif
 }
 
